numpy/lib/function_base.py

```diff
@@ -3964,21 +3966,21 @@ def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
             indices_above = indices_above[:-1]
             n = np.isnan(ap[-1:, ...])
 
-        x1 = take(ap, indices_below, axis=axis) * weights_below
-        x2 = take(ap, indices_above, axis=axis) * weights_above
+        x1 = take(ap, indices_below, axis=axis)
+        x2 = take(ap, indices_above, axis=axis)
 
         # ensure axis with q-th is first
-        x1 = np.moveaxis(x1, axis, 0)
-        x2 = np.moveaxis(x2, axis, 0)
+        offset = subtract(x2, x1) * weights_above
+        offset = np.moveaxis(offset, axis, 0)
 
         if zerod:
             x1 = x1.squeeze(0)
-            x2 = x2.squeeze(0)
+            offset = offset.squeeze(0)
 
         if out is not None:
-            r = add(x1, x2, out=out)
+            r = add(x1, offset, out=out)
```

Comment:
I did not understand what `out` means in this context. If you can give me a hint I might be able to simplify the code even more. I guess there is the possibility to get rid of `x1 = x1.squeeze(0)`

Reply:
In working out what hint to give you, I ended up doing some cleanup of this code myself. Thanks for drawing it to my attention. It might be best for you to wait until my cleanup goes through, and then to rebase your tests.

Reply:
> Thanks for your patience - my cleanup is in. It's unlikely you'll be able to usefully merge the implementation, but you should be able to keep the tests.

Your cleanup made it way easier to implement the montonic lerp. Thanks for that.